### PR TITLE
Loading graph from one or two dataframes

### PR DESCRIPTION
**Pull request by iosonofabio**
As requested in #29, a function to read graphs from one/two dataframes, just like in R.

We are not going to implement the CSV parser anyway, so if this is fine (might need some tests?) then we can add the CSV solutions from @ntamas to the docs (I'll do that) and then close that issue.

### CODE REVIEW

File: tests/test_generators.py [hunk 1]
```diff
@@ -164,6 +164,28 @@ def testWeightedAdjacency(self):
         self.assertTrue(el == [(0,1), (0,2), (1,0), (3,1)])
         self.assertTrue(g.es["w0"] == [1, 2, 2, 1])
 
+    def testDataFrame(self):
```
Comment (ntamas) [inline tests/test_generators.py:167]: Use the `skipIf()` decorator from `tests/utils.py` to skip the test case when `pandas` is not installed. See `tests/test_basic.py` for inspiration; there is a test case that is skipped when NumPy is not installed.

Comment (iosonofabio) [reply tests/test_generators.py:167]: :+1:

File: src/igraph/__init__.py [hunk 2]
```diff
@@ -2850,6 +2850,84 @@ def Incidence(klass, *args, **kwds):
...
+        """
+        import numpy as np
+        import pandas as pd
+
+        if edges.shape[1] < 2:
+            raise ValueError("the data frame should contain at least two columns")
+
+        # Handle if some elements are 'NA'
+        if edges.iloc[:, :2].isna().values.any():
+            warn("In `d' `NA' elements were replaced with string \"NA\"")
```
Comment (ntamas) [inline src/igraph/__init__.py:2876]: what is `d'?

File: src/igraph/__init__.py [hunk 3]
```diff
@@ -2850,6 +2850,84 @@ def Incidence(klass, *args, **kwds):
...
+            edges = edges.copy()
+            edges.iloc[:, :2].fillna('NA', inplace=True)
+
+        if (vertices is not None) and vertices.iloc[:, 0].isna().values.any():
+            warn("In `vertices.iloc[:, 0]` `NA' elements were replaced with string \"NA\"")
```
Comment (ntamas) [inline src/igraph/__init__.py:2881]: I'm not that familiar with Pandas to judge this -- would "vertices.iloc[:, 0]" make sense for an average user that knows Pandas but has no idea about how this function is implemented internally in igraph? Can we come up with a more readable warning?

File: src/igraph/__init__.py (hunk 2 above)
Comment (iosonofabio) [reply src/igraph/__init__.py:2876]: a bad copycat from R ;-) correcting now

File: src/igraph/__init__.py (hunk 3 above)
Comment (iosonofabio) [reply src/igraph/__init__.py:2881]: fair enough, yes I put a more readable warning now

**Comment by iosonofabio** [comment]
I think this can be merged now

**Pull request merged**